examples/cache/cache_control.md

```diff
@@ -0,0 +1,320 @@
+# Claude Prompt Caching Examples
+
+This example demonstrates how to use prompt caching with Claude models across all supported providers through Envoy AI Gateway using the standard `cache_control` API.
+
+## Overview
+
+Envoy AI Gateway supports **provider-agnostic** cache control, allowing you to use the same `cache_control` syntax across all Claude providers:
+
+- **Direct Anthropic** (via Anthropic Messages API)
+- **GCP Vertex AI** (Anthropic Claude models)
+- **AWS Bedrock** (Anthropic Claude models via Converse API)
+
+This unified approach ensures your caching implementation works consistently regardless of the backend provider.
+
+## Benefits
+
+- **Cost Optimization**: Reduce token costs by caching repeated content
+- **Performance**: Faster response times for cached content
+- **Provider Consistency**: Same API across all Claude providers
+- **Token Transparency**: Clear reporting of cache read/write tokens
+- **Easy Migration**: Switch providers without code changes
+
+## Prerequisites
+
+Choose your preferred provider and ensure you have:
+
+### For Direct Anthropic
+
+- Anthropic API key
+- Envoy AI Gateway configured for Anthropic (see [Anthropic setup guide](../../site/docs/getting-started/connect-providers/anthropic.md))
+
+### For GCP Vertex AI
+
+- GCP credentials with Vertex AI access
+- Enabled Claude models in your GCP project
+- Envoy AI Gateway configured for GCP Vertex AI (see [GCP Vertex AI setup guide](../../site/docs/getting-started/connect-providers/gcp-vertexai.md))
+
+### For AWS Bedrock
+
+- AWS credentials with Bedrock access
+- Enabled Claude models in AWS Bedrock
+- Envoy AI Gateway configured for AWS Bedrock (see [AWS Bedrock setup guide](../../site/docs/getting-started/connect-providers/aws-bedrock.md))
+
+## Provider-Specific Considerations
+
+| Feature             | Anthropic Direct | GCP Vertex AI | AWS Bedrock                                                                               |
+| ------------------- | ---------------- | ------------- | ----------------------------------------------------------------------------------------- |
+| Max Cache Points    | No limit         | No limit      | [4 per request](https://docs.aws.amazon.com/bedrock/latest/userguide/prompt-caching.html) |
+| Min Token Threshold | 1,024+ tokens    | 1,024+ tokens | 1,024+ tokens                                                                             |
+| Billing Integration | Native           | Native        | Native                                                                                    |
+| Cache Types         | ephemeral        | ephemeral     | ephemeral                                                                                 |
+
+## Example Requests
+
+### Basic System Prompt Caching
+
+This example works identically across all providers - just change the model name:
+
+```bash
+# Direct Anthropic
+curl -X POST http://localhost:8080/v1/chat/completions \
```

Comment:
This should be using /anthropic/v1/messages endpoint

Reply:
ah good catch ty